mono/utils/mono-threads-coop.c

```diff
@@ -574,34 +572,26 @@ threads_suspend_policy_getenv (void)
 	return policy;
 }
 
-static MonoThreadsSuspendPolicy threads_suspend_policy = -1;
+static char threads_suspend_policy;
 
 static MonoThreadsSuspendPolicy
 mono_threads_suspend_policy (void)
 {
-	MonoThreadsSuspendPolicy policy = threads_suspend_policy;
-	if (G_UNLIKELY (policy == -1)) {
+	int policy = threads_suspend_policy;
+	if (G_UNLIKELY (policy == 0)) {
 		// thread suspend policy:
 		// if the MONO_THREADS_SUSPEND env is set, use it.
 		// otherwise if there's a compiled-in default, use it.
 		// otherwise if one of the old environment variables is set, use that.
 		// otherwise use full preemptive suspend.
-		MonoThreadsSuspendPolicy env_policy = threads_suspend_policy_getenv ();
-		MonoThreadsSuspendPolicy default_policy = threads_suspend_policy_default ();
-		MonoThreadsSuspendPolicy env_compat_policy = threads_suspend_policy_getenv_compat ();
-		if (env_policy)
-			policy = env_policy;
-		else if (default_policy)
-			policy = default_policy;
-		else if (env_compat_policy)
-			policy = env_compat_policy;
-		else
-			policy = MONO_THREADS_SUSPEND_FULL_PREEMPTIVE;
-		
-		g_assert (policy > 0);
-		threads_suspend_policy = policy;
+		   (policy = threads_suspend_policy_getenv ())
+		|| (policy = threads_suspend_policy_default ())
+		|| (policy = threads_suspend_policy_getenv_compat ())
+		|| (policy = MONO_THREADS_SUSPEND_FULL_PREEMPTIVE);
```

Comment:
I like that this tries the assortment of `thread_suspend_policy_*` functions one at a time.
Using short-circuiting `||` instead of `if` is a little unusual for Mono, but alright.

I think in Mono we usually we put the binary operator trailing on the line above:
```c
  a ||
  b ||
  c;
```
but I don't see anything about that in the coding guidelines.

Reply:
Both exist in the wider world. || on left is advertised as more obviously a continuation line. || on the right is popular, and lines up the first line on the left w/o extra artificial indent. clang-tidy for  the win some day...